gbpservice/neutron/services/servicechain/drivers/oneconvergence_intercloud_driver.py

```diff
@@ -141,63 +156,97 @@ def _fetch_template_and_params(self, context, sc_instance,
-                                context._plugin_context, consumer_ptg_subnet_id)
+                        context._plugin_context, consumer_ptg_subnet_id)
                     consumer_cidr = consumer_subnet['cidr']
+                    consumer_port_id, consumer_port_mac = self._get_hotplug_port_id(
+                        context, consumer_ptg_subnet_id, admin_context=admin_context)
                 else:
                     consumer_cidr = '0.0.0.0/0'
+                    consumer_port_id = None
+                provider_subnet = self._core_plugin.get_subnet(
+                                context._plugin_context, provider_subnet_id)
+                provider_cidr = provider_subnet['cidr']
                 stack_template = self._update_template_with_firewall_rules(
                     context, provider_ptg, provider_cidr, consumer_cidr, stack_template,
                     template_aws_version)
 
             filters = {}
             filters['name'] = ["stitching-network"]
-            stitching_networks = self._core_plugin.get_networks(
+            with context._plugin_context.session.begin(subtransactions=True):
+                stitching_networks = self._core_plugin.get_networks(
                                     context._plugin_context, filters)
+            stitching_subnet_id = None
             if stitching_networks:
                 stitching_subnet_id = stitching_networks[0]['subnets'][0]
             else:
                 LOG.error(_("Stitching network is not created"))
                 #TODO(Magesh): Raise exception ??
 
-            (service_instance_id, hotplugged_port_list,
-             stitching_port_ip) = (
-                    self.svc_mgr.create_service_instance(
-                        context._plugin_context, instance_type,
-                        svc_mgmt_pt["port_id"],
-                        provider_subnet_id,
-                        cons_subnet_id=consumer_ptg_subnet_id,
-                        stitching_subnet_id=stitching_subnet_id))
-
-            floating_ip = self.svc_mgr.get_service_instance_mgmt_ip(
-                    context._plugin_context, sc_node['service_type'])
+            stitching_port_id = self._get_stiching_port(admin_context, instance_type, stitching_subnet_id)
+
+            service_instance_id = self.svc_mgr.create_service_instance(
+                            context=context._plugin_context,
+                            tenant_id=context._plugin_context.tenant_id,
+                            service_chain_instance_id=sc_instance_id,
+                            service_type=instance_type,
+                            provider_network_port=provider_port_id,
+                            consumer_network_port=consumer_port_id,
+                            stitching_network_port=stitching_port_id,
+                            management_port=svc_mgmt_pt["port_id"],
+                            mgmt_net_id=svc_mgmt_network_id,
+                            insert_type=insert_type)
```

Comment:
We should move out fip call from service chain driver to the firewall driver to make  it consistent for all services.